build/cmake/FindFmt.cmake

```
@@ -2,15 +2,27 @@ include(ExternalProject)
 
 SET(FMT_VERSION "5.3.0")
 
-ExternalProject_Add(fmt
-    DOWNLOAD_COMMAND git clone --quiet --depth 1 --branch ${FMT_VERSION} --config advice.detachedHead=false https://github.com/DataDog/fmt.git
-    TIMEOUT 5
-    INSTALL_COMMAND ""
-    BUILD_IN_SOURCE TRUE
-    CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE -DFMT_TEST=0 -DFMT_DOC=0
-    BUILD_COMMAND ${CMAKE_COMMAND} -E env CXXFLAGS=-D_GLIBCXX_USE_CXX11_ABI=0 make -j
-    BUILD_BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/fmt-prefix/src/fmt/libfmt.a
-)
+if (ISMACOS)
```

nit: maybe something like this would be better?

```suggestion
if (ISMACOS)
  SET(FMT_CXXFLAGS "-target\ ${OSX_ARCH}-apple-darwin${CMAKE_HOST_SYSTEM_VERSION}\ -D_GLIBCXX_USE_CXX11_ABI=0")
elseif(ISLINUX)
  SET(FMT_CXXFLAGS "-D_GLIBCXX_USE_CXX11_ABI=0")
endif()

ExternalProject_Add(fmt
    DOWNLOAD_COMMAND git clone --quiet --depth 1 --branch ${FMT_VERSION} --config advice.detachedHead=false https://github.com/DataDog/fmt.git
    TIMEOUT 5
    INSTALL_COMMAND ""
    BUILD_IN_SOURCE TRUE
    CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE -DFMT_TEST=0 -DFMT_DOC=0
    BUILD_COMMAND ${CMAKE_COMMAND} -E env CXXFLAGS=${FMT_CXXFLAGS} make -j
    BUILD_BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/fmt-prefix/src/fmt/libfmt.a
)

```

---

done in [bc27402](https://github.com/DataDog/dd-trace-dotnet/pull/3447/commits/bc274021c24a19f561fb286e35a68530bb086679)